ganga/GangaGUI/gui/routes.py

```diff
@@ -64,13 +83,365 @@ def decorated(*args, **kwargs):
+    if int(template_id) not in templates.ids():
+        return jsonify({"success": False,
+                        "message": "Template with ID {} does not exist".format(template_id)}), 400
+
+    # Create job using template
+    try:
+        j = Job(templates[int(template_id)])
+        if job_name is not None:
+            j.name = job_name
+    except Exception as err:
+        return jsonify({"success": False, "message": str(err)})
+
+    return jsonify({"success": True,
+                    "message": "Job with ID {} created successfully using the template ID {}".format(j.id,
+                                                                                                     template_id)})
+
+
+# Perform certain action on the Job - PUT Method
+@app.route("/job/<int:job_id>/<action>", methods=["PUT"])
+@token_required
+def job_action_endpoint(current_user, job_id: int, action: str):
+    """
+    Given the job_id and action in the endpoint, perform the action on the job.
+
+    Supported actions: do_auto_resubmit, name, parallel_submit, copy, kill, force_status, resubmit, runPostProcessors, submit
+
+    Important Info:
+    1) do_auto_resubmit: User needs to provide 'do_auto_resubmit' field and it's value in the request body.
+    2) name: User needs to provide 'name' field and it's value in the request body.
+    3) parallel_submit: User needs to provide 'parallel_submit' field and it's value in the request body.
+    4) force_status: User needs to provide 'force_status' field and it's value in the request body.
+
+    :param job_id: int
+    :param action: str
+    :param attribute: str
+    """
+
+    # Imports
+    from GangaCore.GPI import jobs
+
+    # Job existence check
+    if job_id not in list(jobs.ids()):
+        return jsonify({"success": False, "message": "Job with ID {} does not exist".format(job_id)}), 400
+
+    supported_actions = ["do_auto_resubmit", "name", "parallel_submit", "copy", "kill", "force_status", "resubmit",
+                         "runPostProcessors", "submit"]
+
+    # Action support check
+    if action not in supported_actions:
+        return jsonify({"success": False,
+                        "message": "Job action {} is not currently supported or does not exist".format(action)}), 400
+
+    # Function for action: do_auto_resubmit
+    if action == "do_auto_resubmit":
+
+        # Get value detail from the request data
+        value = request.form.get("do_auto_resubmit")
+
+        # Value check
+        if value is None:
+            return jsonify({"success": False,
+                            "message": "Please provide do_auto_submit value in the request body."}), 400
+
+        # Value validator
+        if value in ["True", "False", "true", "false"]:
+            value = False if value == "False" or value == "false" else True
+        else:
+            return jsonify({"success": False,
+                            "message": "Please provide do_auto_submit value as True or False."}), 400
```

Comment:
This type checking should be factored out to make the code more readable.

Reply:
Or maybe we should not do it at all. Python in general is duck typed and type checking can become very heavy. In this specific case, the only thing you care about is probably if the argument can evaluate to True or False (but it could be an int).

Reply:
I can refactor it, as the value that comes from the client-side is basically of type `str`, and in order to set the value of `do_auto_submit` it needs to be of type `bool`. This is just the type conversion.